### PR TITLE
Update tests for newer docker registry format

### LINKED ISSUE #5142: `bazel run //scenarios:kubernetes_e2e_test` failing locally
Seeing this error:

```
======================================================================
FAIL: test_default_tag (__main__.ScenarioTest)
Ensure the default tag exists on gcr.io.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/justinsb/.cache/bazel/_bazel_justinsb/a1e84e01d786bcdb69b69577a117fc89/execroot/__main__/bazel-out/local-fastbuild/bin/scenarios/kubernetes_e2e_test.runfiles/__main__/scenarios/kubernetes_e2e_test.py", line 397, in test_default_tag
    self.assertNotIn('errors', data)
AssertionError: 'errors' unexpectedly found in {u'errors': [{u'message': u"Manifest with tag 'v20171018-fb8014bc-master' has media type 'application/vnd.docker.distribution.manifest.v2+json', but client accepts '*/*'.", u'code': u'MANIFEST_UNKNOWN'}]}
```

And indeed:
```
> curl https://gcr.io/v2/k8s-testimages/kubekins-e2e/manifests/v20171018-fb8014bc-master
{"errors":[{"code":"MANIFEST_UNKNOWN","message":"Manifest with tag 'v20171018-fb8014bc-master' has media type 'application/vnd.docker.distribution.manifest.v2+json', but client accepts '*/*'."}]}
```

### PR DESCRIPTION
**Pull request by justinsb**
Fix #5142

### CODE REVIEW

**Comment by justinsb** [comment]
Not entirely sure how this isn't failing in CI though...

The V2 response body looks like this, so I check for config.digest:

```
{
   "schemaVersion": 2,
   "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
   "config": {
      "mediaType": "application/vnd.docker.container.image.v1+json",
      "size": 7786,
      "digest": "sha256:ca095d8d6fc275c903e9e612d08bff36714eec6fee0efb4bba280fbc85820a46"
   },
   "layers": [
      {
         "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
         "size": 51476801,
         "digest": "sha256:4a441ce0e165bebd7e27e7059a2ef8a79ca27f0abda6691b3a93d90d4b92e2f1"
      },
      {
         "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
         "size": 96,
         "digest": "sha256:dd7a7fc4af8e05d31955779100831bdf85da8cd6b34de1b29a53cd17538cb3e8"
      },
      {
         "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
         "size": 128239911,
         "digest": "sha256:b17ab39be9b011f344c4e669c86a7166b478c97d45a9b74f0b4b85371e4ab2b1"
      },
      {
         "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
         "size": 77653816,
         "digest": "sha256:e8741b5433cdb4e502fc18418af21201da5ecab70495081b0faa5fc0767b3fb7"
      },
      {
         "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
         "size": 8355106,
         "digest": "sha256:fa2ea612fa21c202fb6b8ddc6e6f9b8483da845397571f795d9324fbaffee982"
      },
      {
         "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
         "size": 693,
         "digest": "sha256:79dab33eea7c04f4cc69ba96cd94e2afffdb029654ecf350e8cdbf10a9e0032b"
      },
      {
         "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
         "size": 89164065,
         "digest": "sha256:62f69d64677152f15fc4ce624b9af1f4ec3322bbb8688b304639de886b047509"
      },
      {
         "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
         "size": 203,
         "digest": "sha256:f6acadd986d910ca2e21d8b7b26958dd91d0eb4693fe9bce39fbc8c559b9cc2c"
      },
      {
         "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
         "size": 170498430,
         "digest": "sha256:ea4d42ca11007bcf7d9befc6dd91c66fa6a7b228f8a037ce3b59c3e392063c44"
      },
      {
         "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
         "size": 3050366,
         "digest": "sha256:853a3c7c6a4973d4c1df6e9400478074ca22fc6cda297244611dd585eb1f0b69"
      },
      {
         "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
         "size": 3063,
         "digest": "sha256:d543cc9b00cfa4491d1ffe64ed7d6484d9050db7be80bd961ddacad4c5cb8987"
      }
   ]
}
```

**Comment by krzyzacy** [comment]
ah thanks, I almost thought that because you don't have access to the image registry

**Comment by krzyzacy** [comment]
/assign
/assign @BenTheElder 
I'll try myself tomorrow :-)

**Comment by justinsb** [comment]
Yeah, definitely non-urgent :-)

File: scenarios/kubernetes_e2e_test.py [hunk 1]
```diff
@@ -393,9 +392,13 @@ def test_default_tag(self):
         self.assertIsNotNone(match)
         url = 'https://gcr.io/v2/%s/manifests/%s' % (match.group(1),
                                                      match.group(2))
-        data = json.loads(urllib.urlopen(url).read())
+        print url
```
Comment (krzyzacy) [inline scenarios/kubernetes_e2e_test.py]: nit: remove this?

**Comment by krzyzacy** [comment]
/lgtm
/hold
feel free to `hold cancel` once you remove the minor nit.
@BenTheElder suspect it works locally for us because we are on Google hardware :-)

File: scenarios/kubernetes_e2e_test.py (hunk 1 above)
Comment (justinsb) [reply scenarios/kubernetes_e2e_test.py]: Yup will do - sorry - forgot to remove my debugging code!

**Comment by justinsb** [comment]
Removed that `print url` line (sorry & good catch)

I still don't understand how the current version passes today in CI though.... any ideas?  Some magic http_proxy or env var?

**Comment by justinsb** [comment]
Idea: we can always hold this and find someone else outside google / GCE to try it!

**Comment by krzyzacy** [comment]
hummm I'll try at home when I get a chance :-)

**Pull request merged**